dhall-nix/src/Dhall/Nix.hs

```diff
@@ -660,4 +703,100 @@ dhallToNix e =
         loop (Dhall.Core.desugarWith a)
     loop (ImportAlt a _) = loop a
     loop (Note _ b) = loop b
-    loop (Embed x) = absurd x
+    loop (Embed (Import (ImportHashed maybeHash importType) mode)) =
+      case importType of
+        Local{} -> return untranslatable
+        Env{} -> return untranslatable
+        Missing -> return untranslatable
+        Remote url ->
+          case mode of
+            Location -> return untranslatable
+            RawText -> return untranslatable
+            Code ->
+              case maybeHash of
+                Nothing -> return untranslatable
+                Just hash ->
+                  -- The following code block creates a function call to
+                  -- dhallToNixUrlFOD that looks like the following:
+                  --
+                  -- dhallToNixUrlFOD
+                  --   (let
+                  --       packageSet =
+                  --         (fetchurl {
+                  --           url = "https://github.com/purescript/package-sets/releases/download/psc-0.14.4-20210905/packages.dhall";
+                  --           hash = "sha256-W1WfXQXIvufW57F5N7fxqRvi/YMBdvlnwlFn4eMyVLc=";
+                  --           downloadToTemp = true;
+                  --           postFetch = "${dhall}/bin/dhall --alpha --plain --file \"\$downloadedFile\" | ${dhall}/bin/dhall encode > \$out";
+                  --         }).overrideAttrs (oldAttrs: {
+                  --           nativeBuildInputs = (oldAttrs.nativeBuildInputs or []) ++ [ cacert ];
+                  --         });
+                  --    in
+                  --      runCommand "decodedPackageSet" {} "${dhall}/bin/dhall decode --file \"${packageSet}\" > \$out"
+                  --   )
+                  --   { inherit dhallToNixUrlFOD cacert dhall fetchurl runCommand; };
+                  return $
+                    "dhallToNixUrlFOD" @@
+                      (letE
+                        "packageSet"
+                        (
+                          ("fetchurl" @@
+                            attrsE
+                              [ ("url", mkStr $ Dhall.Core.pretty url)
+                              , ("hash", mkStr $ dhallHashToSriHash hash)
+                              , ("downloadToTemp", mkBool True)
+                              , ("postFetch",
+                                  mkStrAntiquote
+                                    [ Antiquoted "dhall"
+                                    , Plain "/bin/dhall --alpha --plain --file \"$downloadedFile\" | "
+                                    , Antiquoted "dhall"
+                                    , Plain "/bin/dhall encode > $out"
+                                    ]
```

Comment:
In order to make recursive URL imports work correctly, I setup my `dhallToNixUrlFOD` function to call itself recursively if it encounters a URL import.

Also, in order to make Dhall's hash line-up with Nix's hash, I run it through `dhall --alpha`.

My question is *if I run an expression through `dhall --alpha`, am I guaranteed that all remote imports will be resolved and inlined*?  

If this is true, then I guess I really don't need to worry about calling `dhallToNixUrlFOD` recursively, since there should never be any remote URL imports that need to evaluated recursively.  They should all be resolved during the first call to `dhall --alpha`.